plugin/src/main/kotlin/io/ktor/plugin/KtorGradlePlugin.kt

```diff
@@ -1,40 +1,62 @@
 package io.ktor.plugin
 
 import io.ktor.plugin.features.*
-import io.ktor.plugin.features.KtorExtension.Companion.DEVELOPMENT_MODE_PROPERTY
 import io.ktor.plugin.internal.*
+import io.ktor.plugin.internal.KotlinPluginType.*
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.plugins.ApplicationPlugin
-import org.gradle.api.plugins.JavaApplication
 
 const val KTOR_VERSION = "3.1.2"
 
 @Suppress("unused") // Gradle Plugin is not used directly
 abstract class KtorGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) {
         val extension = project.extensions.create(KtorExtension.NAME, KtorExtension::class.java)
-        configureApplication(project, extension)
+        project.configureApplication(extension)
         configureFatJar(project)
         configureDocker(project)
         configureBomFile(project)
         // Disabled until the native image generation is not possible with a single task with default configs
         // See https://youtrack.jetbrains.com/issue/KTOR-4596/Disable-Native-image-related-tasks
         // configureNativeImage(project)
-    }
 
-    private fun configureApplication(project: Project, extension: KtorExtension) = with(project) {
-        plugins.apply(ApplicationPlugin::class.java)
+        with(project) {
+            var kotlinPluginApplied = false
+            whenKotlinPluginApplied { pluginType ->
+                if (pluginType == Multiplatform) reportKmpCompatibilityWarning()
+                kotlinPluginApplied = true
+            }
 
-        afterEvaluate {
-            if (extension.development.get()) application.enableDevelopmentMode()
+            afterEvaluate {
+                if (!kotlinPluginApplied) reportKotlinPluginMissingWarning()
+            }
         }
     }
-}
 
-private fun JavaApplication.enableDevelopmentMode() {
-    val prefix = "-D$DEVELOPMENT_MODE_PROPERTY="
-    if (applicationDefaultJvmArgs.none { it.startsWith(prefix) }) {
-        applicationDefaultJvmArgs += "${prefix}true"
+    private fun Project.reportKmpCompatibilityWarning() {
+        logger.warn("warning: Ktor Gradle plugin is not fully compatible with Kotlin Multiplatform plugin.")
```

Comment:
Since the log level is specified (`log.warn`), it's not necessary to also include "warning" in the message.

```suggestion
        logger.warn("The Ktor Gradle plugin is not fully compatible with the Kotlin Multiplatform plugin.")
```

Reply:
Unfortunately, Gradle doesn't show log level in logs, so I'm adding it to the message
```suggestion
        logger.warn("warning: The Ktor Gradle plugin is not fully compatible with the Kotlin Multiplatform plugin.")
```